src/java/org/apache/cassandra/index/sai/disk/MemtableTermsIterator.java

```diff
@@ -0,0 +1,134 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.cassandra.index.sai.disk;
+
+import java.nio.ByteBuffer;
+import java.util.Iterator;
+
+import com.google.common.base.Preconditions;
+
+import com.carrotsearch.hppc.LongArrayList;
+import com.carrotsearch.hppc.cursors.LongCursor;
+import org.apache.cassandra.utils.Pair;
+import org.apache.cassandra.utils.bytecomparable.ByteComparable;
+
+/**
+ * Iterator over a token range bounded segment of a Memtable index. Used to flush Memtable index segments to disk.
+ */
+public class MemtableTermsIterator implements TermsIterator
```

Comment:
nit: Seems like the big thing it does is track min/max row ID as it iterates so that we can build segment metadata later. We could name it `InMemoryTermsIterator` perhaps, but we know it's always from a Memtable outside of the tests.

Reply:
It would make more sense to call it `MemtableIndexTermsIterator` but even that doesn't exactly describe what it is doing. I'm inclined not to change it because, as you have said, it's in the `memory` package and it is, fundamentally the terms from a memtable.